vllm/sequence.py

```diff
@@ -550,7 +551,14 @@ def get_seqs(
     ) -> List[Sequence]:
         if status is None:
             return self.seqs
-        return [seq for seq in self.seqs if seq.status == status]
+
+        if self.is_single_seq:
+            if self.seqs[0].status == status:
+                return self.seqs
+            else:
+                return []
+        else:
+            return [seq for seq in self.seqs if seq.status == status]
```

Comment:
Could make this a bit more concise (same for others below):
```suggestion
        if self.is_single_seq:
            return self.seqs if self.seqs[0].status == status else []
        return [seq for seq in self.seqs if seq.status == status]
```